lib/ansible/playbook/role/definition.py

```diff
@@ -228,6 +231,22 @@ def _split_role_params(self, ds):
 
         return (role_def, role_params)
 
+    def _validate_role_name(self, role_name):
+        '''
+        split role_name into parts and validate independently for criteria documented in
+        https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_structure.html#roles-directory and
+        raise a distinct AnsibleError in case of violation
+        '''
+
+        parts = role_name.split(".")
+        for part in parts:
+            # make sure each part starts alphanumerical
+            # remove legal special character before checking if it's alphanumerical
+            # make sure each part is fully lowercase
+            if not part.startswith('_') and not part.replace('_', '').isalnum() and part.islower():
```

Comment:
This [naming restriction](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_structure.html#roles-directory) is specific to importing the role into Galaxy. The requirement just to use a role in a collection is that it needs to be a valid python identifier. That's a requirement for collections too - here's how we validate that: https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/collection_loader/_collection_finder.py#L875.

This fix is missing validation for role names provides as strings, e.g.
```
roles:
  - ns.coll.rolename
```

and as already said, validation should only occur for roles in collections.

Reply:
Thank you! I kind of thought that this is handled somewhere already. I will read into it and update my PR soon. Thanks!